src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -3477,6 +3477,46 @@ void ApplicationManagerImpl::ProcessPostponedMessages(const uint32_t app_id) {
   std::for_each(messages.begin(), messages.end(), push_allowed_messages);
 }
 
+void ApplicationManagerImpl::ProcessOnDataStreamingNotification(
+    const protocol_handler::ServiceType service_type,
+    const uint32_t app_id,
+    const bool streaming_data_available) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  bool should_send_notification = false;
+
+  {
```

Comment:
Any reason why the code below is in a separate block ({...})?

Reply:
@ShobhitAd this extra scope is required for the following `streaming_services_lock_` to be unlocked at the end of the scope, not the entire method.